src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -231,6 +231,26 @@ public async Task<CodeFixCollection> GetDocumentFixAllForIdInSpan(Document docum
             return result.ToImmutableAndFree().FirstOrDefault();
         }
 
+        public async Task<Document> ApplyCodeFixesForSpecificDiagnosticId(Document document, string diagnosticId, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var textSpan = new TextSpan(0, tree.Length);
+
+            var fixCollection = await GetDocumentFixAllForIdInSpan(
+                document, textSpan, diagnosticId, cancellationToken).ConfigureAwait(false);
+            if (fixCollection == null)
+            {
+                return document;
+            }
+
+            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+
+            var solution = await fixAllService.GetFixAllChangedSolutionAsync(
+                fixCollection.FixAllState.CreateFixAllContext(new ProgressTracker(), cancellationToken)).ConfigureAwait(false);
```

Comment:
it seems strange that we're not actually using the ProgressTracker.  The purpose of it is so that we can actually have a real progress bar as we do large amounts of work.